src/main/java/com/prgms/voucher/voucherproject/repository/MemoryVoucherRepository.java

```diff
@@ -0,0 +1,26 @@
+package com.prgms.voucher.voucherproject.repository;
+
+import com.prgms.voucher.voucherproject.domain.Voucher;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class MemoryVoucherRepository implements VoucherRepository{
+    private final Map<UUID, Voucher> storage = new ConcurrentHashMap<>();
+
+    @Override
+    public Optional<Voucher> findById(UUID voucherId) {
+        return Optional.ofNullable(storage.get(voucherId)); // null일 경우 empty 반환
+    }
+
+    @Override
+    public void save(Voucher voucher) {
+        storage.put(voucher.getVoucherId(), voucher);
```

Comment:
id 에 대한 Getter 는 메서드명을 간결하게 유지해도 의미가 있다고 생각해요.
```suggestion
        storage.put(voucher.getId(), voucher);
```